quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -493,19 +486,25 @@ impl Metastore for FileBackedMetastore {
         .await
     }
 
+    // TODO: Switch over to trait defined implemenation once order issue is sorted.
     async fn list_stale_splits(
         &self,
         index_id: &str,
         delete_opstamp: u64,
         num_splits: usize,
     ) -> MetastoreResult<Vec<Split>> {
         self.read(index_id, |index| {
+            let filter = ListSplitsQuery::for_index(index_id)
+                .with_opstamp_older_than(delete_opstamp)
+                .with_limit(num_splits);
+
             let splits = index
-                .list_splits(SplitState::Published, None, None, Some(delete_opstamp))?
+                .list_splits(filter)?
                 .into_iter()
                 .sorted_by_key(|split| split.split_metadata.delete_opstamp)
                 .take(num_splits)
```

Comment:
the `take` is redundant here. Moreover, we apply the ordering after applying the limit... so it can't work :)